metadatastore/commands.py

```diff
@@ -685,21 +1004,18 @@ def find_run_stops(run_start=None, **kwargs):
     _format_time(kwargs)
     # get the actual mongo document
     if run_start:
-        run_start = _get_mongo_document(run_start, RunStart)
-        kwargs['run_start_id'] = run_start.id
-
-    _normalize_object_id(kwargs, '_id')
-    _normalize_object_id(kwargs, 'run_start_id')
-    run_stop = RunStop.objects(__raw__=kwargs).order_by('-time')
-    run_stop = run_stop.no_dereference()
+        run_start = doc_or_uid_to_uid(run_start)
+        run_start = runstart_given_uid(run_start)
+        run_start = _RUNSTART_UID_to_OID_MAP[run_start['uid']]
+        kwargs['run_start_id'] = run_start
 
-    _as_document = _AsDocument()
+    run_stop = RunStop.objects(__raw__=kwargs).as_pymongo()
 
-    return (_as_document(rs) for rs in run_stop)
+    return (_cache_runstop(rs) for rs in run_stop.order_by('-time'))
 
 
 @_ensure_connection
-def find_event_descriptors(run_start=None, **kwargs):
+def find_descriptors(run_start=None, **kwargs):
     """Given search criteria, locate EventDescriptor Documents.
 
     Parameters
```

Comment:
The docstring for the `run_start` parameter needs to be changed to `metadatastore.doc.Document`